rfcs/20210519-single-pass-bnorm/README.md

```diff
@@ -0,0 +1,126 @@
+# Proposal to calculate mean & variance in batchnorm in single pass
+
+## Requester
+GPU Enabling team
+
+## Motivation
+Using new formula for calculating variance will result in up to 33% performance
+improvement in batchnorm fwd training operation. 
+
+Currently bnorm performance is far below projections. Projections are based on
+assumption that bnorm reads input tensor once and writes result once. Current
+implementation requires bnorm to read input tensor 3 times. Proposed change
+will decrease it to 2 reads. There is no known generic algorithm to decrease it
+to 1 read, although such efficiency is possible for special cases where input
+tensor is small enough.
```

Comment:
Do you know how it its implemented in other performance libraries, like cudnn? It would be helpful to understand alternative solutions

Reply:
Intel Parallel Computing Lab/CPU, Habana/Gaudi, Nervana/Crest use faster formula.
CuDNN's batchnorm returns mean and var with imprefect precision that indicates (but not proves) they use it too.

Reply:
Thanks, resolved